### DISCUSSION TITLE
Integration of FEM

### DISCUSSION THREAD

**Discussion by AnastKap** [thliebig/openEMS-Project#113] [Ideas]
Hello,

I am new to the openEMS software and right now I've managed to do the following:

- Create my models and simulate them with the FDTD algorithm the package offers
- Understand the basics of the source code structure and the basic code regarding the naive FDTD algorithm(without MPI and other speed-ups)

My view is to extend the openEMS project and use other solvers, like one based upon FEM. How does that seem as a future extension to the project? Is it worthless or something that can give better simulation results to specific types of problems?

**Comment by thliebig** [comment]
Hi,

I'm not sure that makes too much sense. openEMS is 100% geared to FDTD and "adding" another method would yield little to now benefit to just start fresh.
Of course you can use the CAD engine CSXCAD that is used by openEMS and that describes the geometry and materials and everything. It is literally meant to be used exactly like this and why I always insist that it stays ignorant of openEMS/FDTD as much as possible.
And with CSXCAD 95% of what a typical simulation python or Matlab/Octave file consists of would already be covered. And everything that would be openEMS/FDTD related would need to be different anyways...

So my suggestion: think how an FEM or MoM or whatever tool could look like and use CSXCAD for the geometry and try not to carry the baggage of openEMS which after all is a very very different method than FEM or MoM etc.

But this is just my opinion.. Let me know what you think.
br
Thorsten

**Reply by thliebig** [reply]
I would even recommend to use the CSXCAD python interface and implement a new solver in python directly. Performance critical parts could be implemented at a later stage in C-extensions or even in cython.

At least this is my plan to implement a static solver again if I have some more time available in the future.

**Reply by AnastKap** [reply]
Could you please explain what do you mean by saying "Static Solver"?

**Reply by thliebig** [reply]
A long time ago my I implemented a static fields solver, e.g. electric potential of charged electrodes or with a constant voltage but also for quasi static magnetic fields with currents as source etc.
Someday I want to do that again maybe because it gives good inside (e.g. for teaching) into (static) electric fields and nice animations to see how the electric potential converges...
